src/main/java/io/quarkus/gizmo/BytecodeCreator.java

```diff
@@ -559,4 +571,62 @@ default void breakScope() {
      */
     @Override
     default void close() {}
+    
+    /**
+     * The JVM opcode that can be used for {@link BytecodeCreator#ifValue(ResultHandle, IfOpcode)}.
+     * 
+     * @see Opcodes
+     */
+    enum IfOpcode {
```

Comment:
I don't really like this, you should not need to know about bytecodes to use Gizmo, I think we should add methods for each of them.

Reply:
Well, you need to know something about bytecode (see for example all those `invokeX()` methods, `ifNonZero()` is used to test booleans, etc.). In fact, you need to know a lot about bytecode in order to produce "effective" classes. I can rework the PR though and add those 6 methods.